packages/shorebird_cli/lib/src/validators/shorebird_yaml_asset_validator.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:shorebird_cli/src/commands/init_command.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/validators/validators.dart';
+
+/// Verifies that the shorebird.yaml is found in pubspec.yaml assets.
+class ShorebirdYamlAssetValidator extends Validator {
+  @override
+  String get description => 'shorebird.yaml found in pubspec.yaml assets';
+
+  @override
+  bool canRunInCurrentContext() => shorebirdEnv.hasPubspecYaml;
+
+  @override
+  String get incorrectContextMessage => '''
+The pubspec.yaml file does not exist.
+The command you are running must be run within a Flutter app project.''';
+
+  @override
+  Future<List<ValidationIssue>> validate() async {
+    if (!canRunInCurrentContext()) {
+      return [
+        const ValidationIssue(
+          severity: ValidationIssueSeverity.error,
+          message: 'No pubspec.yaml file found',
+        ),
+      ];
+    }
+
+    if (!shorebirdEnv.pubspecContainsShorebirdYaml) {
```

Comment:
This could be an early return (i.e., `if (shorebirdEnv.pubspecContainsShorebirdYaml) return [];`)